pkg/scheduler/scheduler.go

```diff
@@ -882,3 +802,89 @@ func (s *Scheduler) scaleScheduleClusterResourceBinding(clusterResourceBinding *
 	}
 	return nil
 }
+
+func (s *Scheduler) getTypeFromResourceBindings(ns, name string) interface{} {
+	resourceBinding, err := s.bindingLister.ResourceBindings(ns).Get(name)
+	if errors.IsNotFound(err) {
+		return Unknown
+	}
+
+	if len(resourceBinding.Spec.Clusters) == 0 {
+		return FirstSchedule
+	}
+
+	policyPlacement, policyPlacementStr, err := s.getPlacement(resourceBinding)
+	if err != nil {
+		return Unknown
+	}
+
+	appliedPlacement := util.GetLabelValue(resourceBinding.Annotations, util.PolicyPlacementAnnotation)
+
+	if policyPlacementStr != appliedPlacement {
+		return ReconcileSchedule
+	}
+
+	if policyPlacement.ReplicaScheduling != nil && util.IsBindingReplicasChanged(&resourceBinding.Spec, policyPlacement.ReplicaScheduling) {
+		return ScaleSchedule
+	}
+
+	clusters := s.schedulerCache.Snapshot().GetClusters()
+	for _, tc := range resourceBinding.Spec.Clusters {
+		boundCluster := tc.Name
+		for _, c := range clusters {
+			if c.Cluster().Name == boundCluster {
+				if meta.IsStatusConditionPresentAndEqual(c.Cluster().Status.Conditions, clusterv1alpha1.ClusterConditionReady, metav1.ConditionFalse) {
+					return FailoverSchedule
+				}
+			}
+		}
+	}
+	return nil
```

Comment:
If no match case above, here can be return `AvoidSchedule` according to the original logic.